src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -359,32 +359,6 @@ private static void WriteTo(Metadata? metadata, ObjectWriter writer, Cancellatio
         }
 
         var metadataKind = (MetadataImageKind)imageKind;
-        if (_storageService == null)
-        {
-            if (metadataKind == MetadataImageKind.Assembly)
-            {
-                using var pooledMetadata = Creator.CreateList<ModuleMetadata>();
-
-                var count = reader.ReadInt32();
-                for (var i = 0; i < count; i++)
-                {
-                    metadataKind = (MetadataImageKind)reader.ReadInt32();
-                    Contract.ThrowIfFalse(metadataKind == MetadataImageKind.Module);
-
-#pragma warning disable CA2016 // https://github.com/dotnet/roslyn-analyzers/issues/4985
-                    pooledMetadata.Object.Add(ReadModuleMetadataFrom(reader, kind));
-#pragma warning restore CA2016 
-                }
-
-                return (AssemblyMetadata.Create(pooledMetadata.Object), storages: default);
-            }
-
-            Contract.ThrowIfFalse(metadataKind == MetadataImageKind.Module);
-#pragma warning disable CA2016 // https://github.com/dotnet/roslyn-analyzers/issues/4985
-            return (ReadModuleMetadataFrom(reader, kind), storages: default);
```

Comment:
this block was hte only block that called the two-arg form of ReadModuleMetadataFrom.  So that is then deleted below.